[1.x] Improve test suite

ildyria opened the pull request:
# Description

## Improve yaml for test runs.

- laravel/orchestra version are provided in the composer file (in require-dev).
- make use of https://github.com/ramsey/composer-install to manage the composer installation & cache.
- Now runs on 3 levels:
    - lowest : installs the lowest versions of Composer dependencies   
      (equivalent to running `composer update --prefer-lowest --prefer-stable`)
    - locked : installs the locked versions of Composer dependencies
      (equivalent to running `composer install`)
    - highest : installs the highest versions of Composer dependencies
      (equivalent to running `composer update`)

DarkGhostHunter commented:
1. No need to use locked versions.

- A lowest for old framework versions and dependencies. It uses the locked dependencies as base, that's why its `1.*` instead of a locked version, so it doesn't forces the developer project to update.
- The latest using update, making it future proof.

I don't have a need to test locked versions since the needle can vary between the two extremes.

3. Cache works fine, why not [update to v3](https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows) instead?

2. The `orchestra/testbench` installs the framework automatically, it doesn't need to be required.

ildyria commented:
- Removed the locked version as per requested.
- Removed Laravel as per requested.
- Add support for php 8.2 (to be ready for [November 24, 2022](https://wiki.php.net/todo/php82)).
- Action https://github.com/ramsey/composer-install already makes use of update cache v3 version.

DarkGhostHunter commented:
I'm gonna delay this for now to ensure I can add a more robust test suite like the one in [CacheRefresh](https://github.com/Laragear/CacheRefresh/actions/runs/2725757264).
